File: Day6/Complement of Base 10 Integer/1009_1.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        if n == 0:
            return 1
        
        count = 0
        ans = 0
        
        while n != 0:
            bit = (n & 1) ^ 1
            
            ans += (bit * 2**count)
            
            count += 1
            n = n >> 1
            
        return ans
        

File: Day6/Complement of Base 10 Integer/1009_1.cpp
#include <math.h>

class Solution 
{
    public:
        int bitwiseComplement(int n) 
        {
            int ans = 0;
            int count = 0;
            
            if(n == 0)
            {
                return 1;
            }
            
            while(n!= 0)
            {
                int bit = (n & 1) ^ 1;
                
                ans += (bit * ceil(pow(2, count)));
                
                count++;
                n = n >> 1;
            }
            
            return ans;
        }
};